datafusion-cli/src/object_storage.rs

```diff
@@ -415,6 +424,15 @@ pub(crate) async fn get_object_store(
             let builder = get_oss_object_store_builder(url, options)?;
             Arc::new(builder.build()?)
         }
+        "cos" => {
+            let Some(options) = table_options.extensions.get::<AwsOptions>() else {
```

Comment:
In follow up we need to refactor this part as for different schemas its too much of the common code

Reply:
Yes, agree with you.

Reply:
It would be great to file a ticket to do so 🙏